ddl/ddl_api.go

```diff
@@ -599,6 +599,16 @@ func (d *ddl) buildTableInfo(tableName model.CIStr, cols []*table.Column, constr
 			continue
 		}
 		if constr.Tp == ast.ConstraintPrimaryKey {
+			for _, key := range constr.Keys {
+				col := table.FindCol(cols, key.Column.Name.O)
+				if col == nil {
+					return nil, errKeyColumnDoesNotExits.Gen("key column %s doesn't exist in table", key.Column.Name)
+				}
+				// Virtual columns cannot be used in primary key.
+				if len(col.GeneratedExprString) != 0 && !col.GeneratedStored {
```

Comment:
because stored generated column can be used in primary key. only **virtual generated column** cannot. It's compatible with MySQL. :)